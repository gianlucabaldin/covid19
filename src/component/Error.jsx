import React from 'react';

const Error = () => <div>Ops! An error occured.</div>;

export default Error;
