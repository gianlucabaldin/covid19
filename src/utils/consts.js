export const DEFAUL_MAX_DATA_SIZE = 30;

export const API_ITALY_HYSTORICAL =
  'https://api.covid19api.com/total/dayone/country/italy';

export const API_ITALY_HYSTORICAL_APIFY =
  'https://api.apify.com/v2/datasets/CUdKmb25Z3HjkoDiN/items?format=json&clean=1';
export const API_ITALY_HYSTORICAL_APIFY_SHORT_URL = 'api.apify.com';

export const API_ITALY_REGION =
  'https://openpuglia.org/api/?q=getdatapccovid-19';

export const OPEN_PUGLIA_API = 'openpuglia.org';
export const COVID_19_API = 'covid19api.com';

export const STATUS = {
  CONFIRMED: 'CONFIRMED',
  RECOVERED: 'RECOVERED',
  DEATHS: 'DEATHS',
  ACTIVES: 'ACTIVES',
  INTENSIVE_THERAPY: 'INTENSIVE_THERAPY',
  TOTAL_HOSPITALIZED: 'TOTAL_HOSPITALIZED',
  TOTAL_POSITIVE: 'TOTAL_POSITIVE',
  TAMPONI: 'TAMPONI',
  TOTAL_CASES: 'TOTAL_CASES',
  DECEASED: 'DECEASED',
  NEW_DAILY_POSITIVE: 'NEW_DAILY_POSITIVE',
  DAILY_DECEASED: 'DAILY_DECEASED',
  DAILY_SWABS: 'DAILY_SWABS',
};
