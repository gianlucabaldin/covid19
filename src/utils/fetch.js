import { mockResponseJson } from './mock/mockItalyHistoricalResponse';

export const API_ITALY_HYSTORICAL =
  'https://api.covid19api.com/total/dayone/country/italy';

export const fetch = async (url = API_ITALY_HYSTORICAL, mock = true) => {
  if (mock) {
    return mockResponseJson;
  }
  return fetch(url).then((res) => res.json());
};

export const fetchLastDay = async (mock = true) => {
  if (mock) {
    return mockResponseJson[mockResponseJson.length - 1];
  }
  return fetch().then((res) => res[res.length - 1]);
};
