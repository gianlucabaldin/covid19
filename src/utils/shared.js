/* eslint-disable import/prefer-default-export */

/**
 * Check if the value is a number and more than a thousand, in that case
 * call the toLocalString() witch inserts (eventually) the separator, used
 * in Summary
 * I.E. for italian it inserts the "." ad thousand separator
 * @param {number || string} val the value to insert the separator to if > thousand
 */
export const getLocalizedValue = (val) => {
  if (typeof val === 'number') {
    return Number(val).toLocaleString();
  }
  return val;
};
