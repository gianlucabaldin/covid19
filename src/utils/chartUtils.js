// const DEFAUL_MAX_DATA_SIZE = 20;

import { DEFAUL_MAX_DATA_SIZE } from './consts';

// In order to have a "readable" chart, reduce if too much input values
export const reduceData = (data, maxSize = DEFAUL_MAX_DATA_SIZE) => {
  const { length } = data;
  if (length > maxSize) {
    const y = Math.round(length / maxSize);
    data = data.filter((el, i) => Math.round(i % y) === 0);
  }
  return data;
};
