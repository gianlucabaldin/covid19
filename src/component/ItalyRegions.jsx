import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import MuiTableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import { Box, Hidden, withStyles } from '@material-ui/core';
import { grey } from '@material-ui/core/colors';
import Error from './Error';
import { OPEN_PUGLIA_API } from '../utils/consts';
import DataProvided from './DataProvided';

const useStyles = makeStyles({
  table: {
    minWidth: 650,
  },
  trHeader: {
    backgroundColor: grey[200],
  },
  th: { fontWeight: 700, padding: 0 },
});

const TableCell = withStyles({
  root: {
    border: '1px solid grey',
  },
})(MuiTableCell);

const ItalyRegions = ({ tableData, width }) => {
  const classes = useStyles();

  /** Legend: 
    1- 'ricoverati con sintomi'       hospitalized with symptoms
    2- 'terapia intensiva'            intensive care
    3- 'totale ospedalizzati'         total hospitalized          --> 1 + 2
    4- 'isolamento domiciliare'       home isolation
    5- 'totale positivi'              total positives             --> 3 + 4 
    6- 'variazione totale positivi'   positives variation
    7- 'nuovi positivi'               new positives
    8- 'dimessi guariti'              discharged healed
    9- deceduti                       deaths
    10- 'totale casi'                 discharged healed           --> 5 + 8 + 9
    11- tamponi                       swabs
  */

  if (!tableData || tableData.length === 0) return <Error />;

  // sort alphabetically by region
  tableData.sort((a, b) => (a.regione > b.regione ? 1 : -1));

  return (
    <>
      {/* <Box
       maxHeight={350}
       // width={width || 800}
       style={{ overflowY: 'scroll', paddingBottom: 10 }}
       data-id="italy-regions-box"
     > */}
      <TableContainer component={Paper}>
        <Table
          className={classes.table}
          size="small"
          aria-label="a dense table"
        >
          <TableHead>
            <TableRow key="header" className={classes.trHeader}>
              <TableCell align="center">Region</TableCell>
              <Hidden lgDown>
                <TableCell align="center">Hospitalized With Symptoms</TableCell>
              </Hidden>
              <TableCell align="center">Intensive Care</TableCell>
              <Hidden lgDown>
                <TableCell align="center">Total Hospitalized</TableCell>
              </Hidden>
              <Hidden mdDown>
                <TableCell align="center">Home Isolation</TableCell>
              </Hidden>
              <TableCell align="center">Total Positives</TableCell>
              <Hidden lgDown>
                <TableCell align="center">Positives Variation</TableCell>
              </Hidden>
              <Hidden lgDown>
                <TableCell align="center">New Positives</TableCell>
              </Hidden>
              <Hidden lgDown>
                <TableCell align="center">Discharged Healed</TableCell>
              </Hidden>
              <TableCell align="center">Deaths</TableCell>
              <TableCell align="center">Total Cases</TableCell>
              <Hidden mdDown>
                <TableCell align="center">Swabs</TableCell>
              </Hidden>
            </TableRow>
          </TableHead>
          <TableBody>
            {tableData &&
              tableData.length > 0 &&
              tableData.map((row) => (
                <TableRow key={row.regione} hover>
                  <TableCell align="center">{row.regione}</TableCell>
                  <Hidden lgDown>
                    <TableCell align="center">
                      {row['ricoverati con sintomi']}
                    </TableCell>
                  </Hidden>
                  <TableCell align="center">
                    {row['terapia intensiva']}
                  </TableCell>
                  <Hidden lgDown>
                    <TableCell align="center">
                      {row['totale ospedalizzati']}
                    </TableCell>
                  </Hidden>
                  <Hidden mdDown>
                    <TableCell align="center">
                      {row['isolamento domiciliare']}
                    </TableCell>
                  </Hidden>
                  <TableCell align="center">{row['totale positivi']}</TableCell>
                  <Hidden lgDown>
                    <TableCell align="center">
                      {row['variazione totale positivi']}
                    </TableCell>
                  </Hidden>
                  <Hidden lgDown>
                    <TableCell align="center">
                      {row['nuovi positivi']}
                    </TableCell>
                  </Hidden>
                  <Hidden lgDown>
                    <TableCell align="center">
                      {row['dimessi guariti']}
                    </TableCell>
                  </Hidden>
                  <TableCell align="center">{row.deceduti}</TableCell>
                  <TableCell align="center">{row['totale casi']}</TableCell>
                  <Hidden mdDown>
                    <TableCell align="center">{row.tamponi}</TableCell>
                  </Hidden>
                </TableRow>
              ))}
          </TableBody>
        </Table>
      </TableContainer>

      <DataProvided date={tableData[0].data} href={OPEN_PUGLIA_API} />
      {/* </Box> */}
    </>
  );
};
export default ItalyRegions;
