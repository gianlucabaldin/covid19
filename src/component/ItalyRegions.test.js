import React from 'react';
import { shallow } from 'enzyme';
import { TableContainer } from '@material-ui/core';
import ItalyRegions from './ItalyRegions';
import { fetchItalyRegion } from '../utils/fetch';
import Error from './Error';

describe('ItalyRegions', () => {
  let wrapper = shallow(<ItalyRegions />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders an error if no data aare passed', () => {
    expect(wrapper.find(Error).length).toEqual(1);
  });

  it('renders an error if empty data are passed and shows error', async () => {
    wrapper = shallow(<ItalyRegions tableData={[]} />);
    expect(wrapper.find('Error').length).toEqual(1);
  });

  describe('fetches', () => {
    it('fetches data and show them within the table', async () => {
      const tableData = await fetchItalyRegion();
      wrapper = shallow(<ItalyRegions tableData={tableData} />);
      expect(wrapper).toIncludeText('Piemonte');
    });
  });
});
