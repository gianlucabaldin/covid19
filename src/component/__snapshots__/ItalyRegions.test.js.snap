// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItalyRegions renders properly 1`] = `<Error />`;
